### DISCUSSION TITLE
Spotify album / playlist links, some songs playing on Youtube instead of Spotify

### DISCUSSION THREAD

**Discussion by ByAhri** [TopiSenpai/LavaSrc#33] [Q&A] [answered]
I haven't figure out to Not to play any youtube song even if I'm only using spotify, and only play Youtube giving youtube URLs,
If I disable youtube completely from lavalink application.yml , then songs stop after about 30 secs, I'm assuming cause it's not playing them on Spotify and then using SoundCloud instead. disabling soundcloud and youtube together just makes it not able to olay any sound at all. I wonder how some bots strictly play Spotify only, and SoundCloud only when SoundCloud URLs are given. 
I have to point out that I set erela.js default search method to Spotify. and with normal query like song name, it does looks for spotify,
it gives me Spotify URI's so Spotify as default search method Works, but the Audio itself is the one that fails to play Spotify tracks when I give Spotify album / Playlists URLs. It almost never happens when I give tracks 1 by 1 though.

I just wanna know for some way to the tracks Audio / files strictly be from Spotify

Edit: Spotify can't be used as search provider I tried to do that using spsearch on lavasrc: providers:
so idk what to do at this point

**Comment by topi314** [comment, marked as answer]
as mentioned in the readme spotify does not actually play from spotify
instead it just resolves spotify links and plays them from different platforms like youtube or soundcloud.
So if you would use `spsearch:` as a provider you would end up in a infinite loop.

You can configure any other source like deezer to resolve spotify tracks but it's not possible to play from spotify directly
their drm & restrictions are too annoying

**Reply by ByAhri** [reply]
I see, I think for now I can change the string query to only look for official audios, I know that's not your concern or that u don't provide that content but do you know how could I get a deezer decryption key or the same for the other platforms that your plugin supports ? so I could use other sources since eventually it won't be legal that I use YT if I want a verified application. and thank U so much for your reply ♡

**Reply by topi314** [reply]
you can find that the deezer key in various other github repos
google is your friend

as for other platforms this plugin supports there aren't any except yandex which has a tutorial in the readme (apple music works the same as spotify)

and no problem

**Reply by ByAhri** [reply]
so deezer would be the same case as spotify and apple or you can actually play deezer ?

**Reply by topi314** [reply]
Deezer actually plays from deezer (their audio needs decryption tho that's why you need the deezer master key thingy)

Apple Music & Spotify try to get the music from your configured sources

**Reply by ByAhri** [reply]
is it possible to do  "ytmsearch:%QUERY%" for youtube music instead of youtube in the plugin providers ?

**Reply by ByAhri** [reply]
ytmsearch seems to work

**Reply by topi314** [reply]
you can use anything you want except spotify & apple music
although I would recommend you do `"ytmsearch:%ISRC%"` first for more exact results
